.gitignore

```diff
@@ -225,3 +225,7 @@ Examples/r/*/.RData
 Examples/test-suite/scilab/*/
 loader.sce
 
+vcpkg.json
+CMakePresets.json
+.cache
+.vscode
```

Comment:
These changes are unrelated to the stated purpose of the PR - if you want to propose adding these, please make that a separate PR.

Reply:
I don't know what all of these are from, but at least `.vscode` is really better handled by configuring it for your own user account as then it'll be ignored in any git repo you work on.  Otherwise SWIG (and every other project) ends up having to add ignore rules for the backup files, etc created by every possible editor and IDE people might use...